components/telemetry-operator/controllers/telemetry/logparser_controller.go

```diff
@@ -122,6 +116,28 @@ func (r *LogParserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	return ctrl.Result{}, nil
 }
 
+func (r *LogParserReconciler) updateCR(ctx context.Context, updateCR bool, logParser telemetryv1alpha1.LogParser) error {
```

Comment:
Can we call it `updateLogParser` (and `updateLogPipeline` in the other controller)? The meaning of the shortcut CR is not very obvious.
Maybe we should also indicate that the behaviour is conditional: `updateLogParserIfChanged`